### PR TITLE
Button component test

### PR DESCRIPTION
**Pull request by williamdaconceicao**
Create a test for the Button component

### CODE REVIEW

**Comment by willdurand** [comment]
@williamdaconceicao nice start, before moving to the next component, please add test cases for most of the component behaviors, i.e. add test cases for each prop: https://github.com/Zoapp/react-materialcomponents/blob/d508b8c69890376c8a4ee2994e865cd0d49f067c/src/libs/components/button.jsx#L96-L112.

File: tests/components/button.test.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import Button from "@libs/components/button";
+
+describe("components/Button", () => {
+  it("renders correctly", () => {
+  	const click = () => console.log("You've clicked");
+    let component = renderer.create(<Button children="Click me" />).toJSON();
+    expect(component).toMatchSnapshot();
```
Comment (willdurand) [inline tests/components/button.test.js]: Could you create different test cases with `it(...)`?

It should be:

```
it("can be unelevated", ...)
it("can be dense", ...)
it("can be compact", ...)
it("renders an icon if supplied", ...)
it("renders a link if supplied", ...)
```

File: tests/components/button.test.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    component = renderer.create(<Button icon="favorite" />).toJSON();
+    expect(component).toMatchSnapshot();
+    component = renderer.create(<Button link="#" />).toJSON();
+    expect(component).toMatchSnapshot();
+    component = renderer.create(<Button ripple />).toJSON();
+    expect(component).toMatchSnapshot();
+    component = renderer.create(<Button cardAction />).toJSON();
+    expect(component).toMatchSnapshot();
+    component = renderer.create(<Button onClick={click()} />).toJSON();
```
Comment (willdurand) [inline tests/components/button.test.js]: You dont have to test this prop.

File: tests/components/button.test.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import Button from "@libs/components/button";
+
+describe("components/Button", () => {
+  let component;
```
Comment (willdurand) [inline tests/components/button.test.js]: this should not be there. use `const component` in each `it` block

File: tests/components/button.test.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  it("can be childrened", () => {
+    component = renderer.create(<Button children="Click me" />).toJSON();
+    expect(component).toMatchSnapshot();
+  });
+  it("can be raised.", () => {
```
Comment (willdurand) [inline tests/components/button.test.js]: add

blank

lines

between

your

code

blocks.

Comment (williamdaconceicao) [reply tests/components/button.test.js]: Done

File: tests/components/button.test.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import renderer from "react-test-renderer";
+import Button from "@libs/components/button";
+
+describe("components/Button", () => {
+  it("can be childrened", () => {
+    const component = renderer.create(<Button children="Click me" />).toJSON();
+    expect(component).toMatchSnapshot();
+  });
+  
+  it("can be raised.", () => {
```
Comment (willdurand) [inline tests/components/button.test.js]: not period needed

File: tests/components/button.test.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import Button from "@libs/components/button";
+
+describe("components/Button", () => {
+  it("can be childrened", () => {
```
Comment (willdurand) [inline tests/components/button.test.js]: it can have children.

File: tests/components/button.test.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    const component = renderer.create(<Button children="Click me" />).toJSON();
```
Comment (willdurand) [inline tests/components/button.test.js]: that's not how we pass `children` prop in React. This is actually a special prop that means "things inside the component":

```js
<Button>
  children are here!
</Button>
```

File: tests/components/button.test.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    const component = renderer.create(<Button unelevated />).toJSON();
+    expect(component).toMatchSnapshot();
+  });
+
+  it("can be stroked", () => {
+    const component = renderer.create(<Button stroked />).toJSON();
+    expect(component).toMatchSnapshot();     
+  });
+
+  it("can be densed", () => {
```
Comment (willdurand) [inline tests/components/button.test.js]: s/densed/dense

File: tests/components/button.test.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    const component = renderer.create(<Button dense />).toJSON();
+    expect(component).toMatchSnapshot();
+  });
+
+  it("can be compacted", () => {
```
Comment (willdurand) [inline tests/components/button.test.js]: s/compacted/compact

File: tests/components/button.test.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    const component = renderer.create(<Button compact />).toJSON();
+    expect(component).toMatchSnapshot();
+  });
+
+  it("can be secondary", () => {
```
Comment (willdurand) [inline tests/components/button.test.js]: "a secondary button"

File: tests/components/button.test.js [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    const component = renderer.create(<Button secondary />).toJSON();
+    expect(component).toMatchSnapshot();
+  });
+
+  it("can be iconed", () => {
```
Comment (willdurand) [inline tests/components/button.test.js]: "can render an icon if supplied"

File: tests/components/button.test.js [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    const component = renderer.create(<Button icon="favorite" />).toJSON();
+    expect(component).toMatchSnapshot();
+  });
+
+  it("can be linked", () => {
```
Comment (willdurand) [inline tests/components/button.test.js]: "can render a link if supplied"

File: tests/components/button.test.js [hunk 13]
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import Button from "@libs/components/button";
+
+describe("components/Button", () => {
+  it("can have children", () => {
+    const component = renderer.create(<Button children="Children are here" />).toJSON();
```
Comment (willdurand) [inline tests/components/button.test.js]: this one needs to be updated.

File: tests/components/button.test.js [hunk 14]
```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import Button from "@libs/components/button";
+
+describe("components/Button", () => {
+  it("can have children", () => {
+    const component = renderer.create(
+      <Button>
+      Children are here!
```
Comment (willdurand) [inline tests/components/button.test.js]: this should be indented

File: tests/components/button.test.js [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+    const component = renderer.create(<Button dense />).toJSON();
+    expect(component).toMatchSnapshot();
+  });
+
+  it("can be compact", () => {
+    const component = renderer.create(<Button compact />).toJSON();
+    expect(component).toMatchSnapshot();
+  });
+
+  it("a secondary button", () => {
```
Comment (willdurand) [inline tests/components/button.test.js]: "it renders a secondary button"

**Comment by willdurand** [comment]
@williamdaconceicao looks good, you can squash your commits now

**Pull request merged**